packages/components/inputs/input-utils/src/multiline-input/multiline-input.tsx

```diff
@@ -1,31 +1,53 @@
-import React from 'react';
-import PropTypes from 'prop-types';
-import requiredIf from 'react-required-if';
-import TextareaAutosize from 'react-textarea-autosize';
+import React, { ChangeEventHandler } from 'react';
+import TextareaAutosize, {
+  TextareaHeightChangeMeta,
+} from 'react-textarea-autosize';
 import { filterDataAttributes } from '@commercetools-uikit/utils';
 import { getTextareaStyles } from './multiline-input.styles';
 
 const MIN_ROW_COUNT = 1;
 
-const MultilineInput = (props) => {
+export type TMultiLineInputProps = {
+  autoComplete?: string;
+  className?: string;
+  hasError?: boolean;
+  hasWarning?: boolean;
+  id?: string;
+  isAutofocussed?: boolean;
+  isDisabled?: boolean;
+  isReadOnly?: boolean;
+  name?: string;
+  onBlur?: ChangeEventHandler;
+  onChange?: ChangeEventHandler;
+  onFocus?: ChangeEventHandler;
+  placeholder?: string;
+  value: string;
+  isOpen: boolean;
+  onHeightChange?: (height: number, rowCount: number) => void;
+};
+
+const MultilineInput = (props: TMultiLineInputProps) => {
   const { onHeightChange } = props;
   const ref = React.useRef();
-  const handleHeightChange = React.useCallback(
-    (height, meta) => {
-      const rowCount = Math.floor(ref.current.scrollHeight / meta.rowHeight);
-      onHeightChange(height, rowCount);
+  const handleHeightChange = React.useCallback<
+    (height: number, meta?: TextareaHeightChangeMeta) => void
+  >(
+    (height: number, meta?: TextareaHeightChangeMeta) => {
+      // @ts-ignore
+      const rowCount = Math.floor(ref?.current?.scrollHeight / meta?.rowHeight);
```

Comment:
I'm still unsure how to address this, so I added an `ignore`
![image](https://user-images.githubusercontent.com/462507/115015099-77224000-9eb3-11eb-9407-f1844be43af0.png)

Reply:
Have you tried typing `useRef`?

Reply:
aah okay. I applied a type before but still got an error. This error is gone if I set the `scrollHeight` default value to `0`
Have a look at the diff. if we accept this, I'll proceed

<details>
 <summary>DIFF</summary>

```diff
diff --git a/packages/components/inputs/input-utils/src/multiline-input/multiline-input.tsx b/packages/components/inputs/input-utils/src/multiline-input/multiline-input.tsx
index 02da6c4f..75c3c537 100644
--- a/packages/components/inputs/input-utils/src/multiline-input/multiline-input.tsx
+++ b/packages/components/inputs/input-utils/src/multiline-input/multiline-input.tsx
@@ -20,39 +20,43 @@ export type TMultiLineInputProps = {
   name?: string;
   onBlur?: ChangeEventHandler;
   onChange?: ChangeEventHandler;
   onFocus?: ChangeEventHandler;
   placeholder?: string;
   value: string;
   isOpen: boolean;
   onHeightChange?: (height: number, rowCount: number) => void;
 };
 
+type TInputRef = {
+  scrollHeight: number;
+};
+
 const MultilineInput = (props: TMultiLineInputProps) => {
   const theme = useTheme();
   const { onHeightChange } = props;
-  const ref = React.useRef();
+
+  const ref = React.useRef<TInputRef>({ scrollHeight: 0 });
   const handleHeightChange = React.useCallback<
-    (height: number, meta?: TextareaHeightChangeMeta) => void
+    (height: number, meta: TextareaHeightChangeMeta) => void
   >(
-    (height: number, meta?: TextareaHeightChangeMeta) => {
-      // @ts-ignore
-      const rowCount = Math.floor(ref?.current?.scrollHeight / meta?.rowHeight);
+    (height: number, meta: TextareaHeightChangeMeta) => {
+      const rowCount = Math.floor(ref.current.scrollHeight / meta.rowHeight);
       if (onHeightChange) {
         onHeightChange(height, rowCount);
       }
     },
     [ref, onHeightChange]
   );
+
   return (
     <TextareaAutosize
-      // @ts-ignore
       ref={ref}
       name={props.name}
       onHeightChange={handleHeightChange}
       autoComplete={props.autoComplete}
       value={props.value}
       onChange={props.onChange}
       id={props.id}
       onBlur={props.onBlur}
       onFocus={props.onFocus}
       disabled={props.isDisabled}
```

</details>